projects/packages/forms/src/contact-form/class-contact-form-plugin.php

```diff
@@ -570,6 +570,159 @@ public static function block_attributes_to_shortcode_attributes( $atts, $type, $
 		return $atts;
 	}
 
+		/**
+		 * Render the number field.
+		 *
+		 * @param array  $atts - the block attributes.
+		 * @param string $content - html content.
+		 *
+		 * @return string HTML for the number field.
+		 */
+	public static function gutenblock_render_form_step( $atts, $content ) {
+		static $step = 0;
+		++$step;
+
+		$version = defined( 'JETPACK__VERSION' ) ? \JETPACK__VERSION : '0.1';
+
+		\wp_enqueue_script_module(
+			'jetpack-form-step',
+			plugins_url( '../../dist/modules/form-step/view.js', __FILE__ ),
+			array( '@wordpress/interactivity' ),
+			$version
+		);
+
+		// Process content for marker classes and add interactivity
+		$processed_content = $content;
+
+		// Only process if we have the WP_HTML_Tag_Processor
+		if ( class_exists( 'WP_HTML_Tag_Processor' ) ) {
+			$blocks_content = do_blocks( $content );
+			$tags           = new \WP_HTML_Tag_Processor( $blocks_content );
+
+			// Process blocks with the "next step" trigger
+			while ( $tags->next_tag( array( 'class_name' => 'trigger-next-step' ) ) ) {
+				// No need to set data-wp-interactive since the parent div already has it
+				$tags->set_attribute( 'data-wp-on--click', 'actions.nextStep' );
+			}
+
+			// Reset and process blocks with the "previous step" trigger
+			$tags->set_bookmark( 'start' );
+			$tags->seek( 'start' );
+			while ( $tags->next_tag( array( 'class_name' => 'trigger-previous-step' ) ) ) {
+				$tags->set_attribute( 'data-wp-on--click', 'actions.previousStep' );
+			}
+
+			$processed_content = $tags->get_updated_html();
+		} else {
+			$processed_content = do_blocks( $content );
+		}
+
+		return '<div data-wp-interactive="jetpack/form" class="jetpack-form-step" data-wp-class--is-before-current="state.isBeforeCurrent" data-wp-class--is-after-current="state.isAfterCurrent" data-wp-class--is-current-step="state.isCurrentStep" ' . wp_interactivity_data_wp_context( array( 'step' => $step ) ) . ' >'
+				. $processed_content
+			. '</div>';
+	}
+
+	/**
+	 * Render the number field.
+	 *
+	 * @param array  $atts - the block attributes.
+	 * @param string $content - html content.
+	 *
+	 * @return string HTML for the number field.
+	 */
+	public static function gutenblock_render_form_step_navigation( $atts, $content ) {
+
+		$version = defined( 'JETPACK__VERSION' ) ? \JETPACK__VERSION : '0.1';
```

Comment:
I've seen the constant manager package used but really no biggie either way, just offering in case you prefer:

```php
Constants::get_constant( 'JETPACK__VERSION' )
```

Reply:
👍